src/xdplwf/recv.c

```diff
@@ -1726,6 +1727,42 @@ XdpGenericReceiveInspect(
             RxQueue->FrameRing->ConsumerIndex, RxQueue->FrameRing->InterfaceReserved,
             NbHead, NextNb);
 
+        if (RxQueue->Flags.ChecksumOffloadEnabled) {
```

Comment:
This logic should be moved into `XdpGenericReceivePreinspectNb` within `XdpGenericReceivePreInspectNbs` I believe. This handles all the loop complexity.

Reply:
Done.